src/Microsoft.Health.Fhir.Core/Features/Operations/BulkDelete/BulkDeleteOrchestratorJob.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.BulkDelete;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Api.Features.Operations.BulkDelete
+{
+    [JobTypeId((int)JobType.BulkDeleteOrchestrator)]
+    public class BulkDeleteOrchestratorJob : IJob
+    {
+        private IQueueClient _queueClient;
+        private ISearchService _searchService;
+
+        public BulkDeleteOrchestratorJob(
+            IQueueClient queueClient,
+            ISearchService searchService)
+        {
+            EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            EnsureArg.IsNotNull(searchService, nameof(searchService));
+
+            _queueClient = queueClient;
+            _searchService = searchService;
+        }
+
+        public async Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progress, CancellationToken cancellationToken)
```

Comment:
Should we add logging mentioning the state of the job? 
I believe it'll help us debugging problems in prod.

Reply:
Good idea. I'm adding it to JobHosting so it is generic for all job types.